site/src/Map.jsx

```diff
@@ -1,135 +1,313 @@
+        />
+      ) : null}
+      {extrusion ? (
+        <Layer
+          filter={[
+            "all",
+            ["==", ["geometry-type"], "Polygon"],
+            ["!=", ["get", "has_parts"], true],
+          ]} // prevent z-fighting
+          beforeId="divisions_division"
+          id={`${theme}_${type}_fill-extrusion`}
+          type="fill-extrusion"
+          source={theme}
+          source-layer={type}
+          paint={{
+            "fill-extrusion-color": color,
+            "fill-extrusion-opacity": 0.35,
+            "fill-extrusion-base": ["get", "min_height"],
+            "fill-extrusion-height": ["get", "height"],
+          }}
+        />
+      ) : null}
+    </>
+  );
 };
-export default function Map({mode}) {
 
-  const [pmTilesReady, setPmTilesReady] = useState(false)
-  const [cursor, setCursor] = useState('auto');
+ThemeTypeLayer.propTypes = {
+  theme: PropTypes.string.isRequired,
+  type: PropTypes.string.isRequired,
+  color: PropTypes.string.isRequired,
+  point: PropTypes.bool,
+  line: PropTypes.bool,
+  polygon: PropTypes.bool,
+  extrusion: PropTypes.bool,
+};
+
+export default function Map({ mode }) {
+  const mapRef = useRef();
+  const [cursor, setCursor] = useState("auto");
   const [mapEntity, setMapEntity] = useState({});
 
+  const [visibleThemes, setVisibleThemes] = useState([]);
+  const [interactiveLayerIds, setInteractiveLayerIds] = useState([]);
+
   useEffect(() => {
-    const protocol = new pmtiles.Protocol()
-    maplibregl.addProtocol('pmtiles', protocol.tile)
-    setPmTilesReady(true)
+    const protocol = new pmtiles.Protocol();
+    maplibregl.addProtocol("pmtiles", protocol.tile);
+    return () => {
+      maplibregl.removeProtocol("pmtiles");
+    };
   }, []);
 
-  const [interactiveLayerIds, setInteractiveLayerIds] = useState(['places']);
-  // const onInteractiveLayersChange = useCallback(layerFilter => {
-  //   setInteractiveLayerIds(MAP_STYLE.layers.map(layer => layer.id).filter(layerFilter));
-  // }, []);
+  const syncInteractiveLayerIds = useCallback(() => {
+    const layers = mapRef.current.getStyle().layers;
+    const layersToShow = layers
+      .filter((layer) => visibleThemes.indexOf(layer.source) >= 0)
+      .map((layer) => layer.id);
+    setInteractiveLayerIds(layersToShow);
+  }, [visibleThemes]);
 
-  const onMouseEnter = useCallback(() => setCursor('pointer'), []);
-  const onMouseLeave = useCallback(() => setCursor('auto'), []);
+  const onMouseEnter = useCallback(() => setCursor("pointer"), []);
+  const onMouseLeave = useCallback(() => setCursor("auto"), []);
 
-  const onClick = useCallback(event => {
+  const onClick = useCallback((event) => {
     const feature = event.features && event.features[0];
-
     if (feature) {
-      setMapEntity(event.features[0].properties);
+      setMapEntity({
+        theme: feature.source,
+        type: feature.sourceLayer,
+        ...feature.properties,
+      });
```

Comment:
Now that we have more layers, we can't assume there will only be one active map entity at a time.

We discussed two options:
1. A Maputnik style popup where you require two clicks to active the inspector panel: (a) click the map and display popup with every feature selected, (b) click the feature you want to inspect.
2. A separate radio button control for which layer should display in the inspector panel on click.

After some thought, I think #1 is a good starting point as it negates the need for an additional control and enables quick inspection of all features at a given lat/lng. Any thoughts/opinions on this?

No need to add as part of PR - I'm planning on adding this functionality once this PR lands.

Reply:
1) is necessary even with 2) because `building_part` and `division`, even if viewed in isolation, will always have multiple overlapping entities at the same position.